server/cmt_cmds.go

```diff
@@ -249,3 +258,96 @@ $ %s query block --%s=%s <hash>
 
 	return cmd
 }
+
+func BootstrapStateCmd(appCreator types.AppCreator) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "bootstrap-state",
+		Short: "Bootstrap CometBFT state at an arbitrary block height using a light client",
+		Long: `
+Bootstrap CometBFT state at an arbitrary block height using a light client
+`,
+
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			serverCtx := GetServerContextFromCmd(cmd)
+			logger := log.NewLogger(cmd.OutOrStdout())
+			cfg := serverCtx.Config
+
+			height, err := cmd.Flags().GetInt64("height")
+			if err != nil {
+				return err
+			}
+			if height == 0 {
+				home := serverCtx.Viper.GetString(flags.FlagHome)
+				db, err := openDB(home, GetAppDBBackend(serverCtx.Viper))
+				if err != nil {
+					return err
+				}
+
+				app := appCreator(logger, db, nil, serverCtx.Viper)
+				height = app.CommitMultiStore().LastCommitID().Version
+			}
+
+			blockStoreDB, err := node.DefaultDBProvider(&node.DBContext{ID: "blockstore", Config: cfg})
+			if err != nil {
+				return err
+			}
+			blockStore := store.NewBlockStore(blockStoreDB)
+
+			stateDB, err := node.DefaultDBProvider(&node.DBContext{ID: "state", Config: cfg})
+			if err != nil {
+				return err
+			}
+			stateStore := sm.NewStore(stateDB, sm.StoreOptions{
+				DiscardABCIResponses: cfg.Storage.DiscardABCIResponses,
+			})
+
+			genState, _, err := node.LoadStateFromDBOrGenesisDocProvider(stateDB, node.DefaultGenesisDocProviderFunc(cfg))
+			if err != nil {
+				return err
+			}
+
+			stateProvider, err := statesync.NewLightClientStateProvider(
+				cmd.Context(),
+				genState.ChainID, genState.Version, genState.InitialHeight,
+				cfg.StateSync.RPCServers, light.TrustOptions{
+					Period: cfg.StateSync.TrustPeriod,
+					Height: cfg.StateSync.TrustHeight,
+					Hash:   cfg.StateSync.TrustHashBytes(),
+				}, servercmtlog.CometLoggerWrapper{Logger: logger.With("module", "light")})
+			if err != nil {
+				return fmt.Errorf("failed to set up light client state provider: %w", err)
+			}
+
+			state, err := stateProvider.State(cmd.Context(), uint64(height))
+			if err != nil {
+				return fmt.Errorf("failed to get state: %w", err)
+			}
+
+			commit, err := stateProvider.Commit(cmd.Context(), uint64(height))
+			if err != nil {
+				return fmt.Errorf("failed to get commit: %w", err)
+			}
+
+			if err := stateStore.Bootstrap(state); err != nil {
+				return fmt.Errorf("failed to bootstrap state: %w", err)
+			}
+
+			if err := blockStore.SaveSeenCommit(state.LastBlockHeight, commit); err != nil {
+				return fmt.Errorf("failed to save seen commit: %w", err)
+			}
+
+			store.SaveBlockStoreState(&cmtstore.BlockStoreState{
+				// maintain the invariant that blocks in range [Base, Height] exists.
+				Base:   state.LastBlockHeight + 1,
```

Comment:
```suggestion
				Base:   state.LastBlockHeight,
```

`Base > Height` will be rejected in peer negotiation